packages/KTable/KTable.vue

```diff
@@ -757,6 +757,12 @@ export default defineComponent({
       offset.value = previousOffset.value
     }
 
+    const shouldShowPagination = computed(() => {
+      return props.fetcher && !props.disablePagination &&
+        !(props.paginationType !== 'offset' && props.hidePaginationWhenOptional && total.value <= pageSize.value) &&
+        !(props.paginationType === 'offset' && props.hidePaginationWhenOptional && !previousOffset.value && !offset.value)
+    })
```

Comment:
🙏🏼  _Just because I this is definitely complicated_...  can you either add a block comment above explaining the logic, or split it into multiple lines and add comments? Future us will forever be in your debt